### ISSUE TITLE
ResourceHistogram add resource tree support

### ISSUE THREAD

**Issue by arcady-zherdev** [bryntum/support#2608]
So resource groups would aggregate child resources allocation.

Requested here: https://www.bryntum.com/forum/viewtopic.php?f=52&t=17095&p=84994

**Comment by arcady-zherdev** [comment]
There are three alternative approaches 

I. The simplest one. We can subclass `grouping` feature for the histogram so it would aggregate and display allocation on group level for a custom grouping criteria.

II. Add ability to assign more than 100% of a resource (we have a ticket for that: https://github.com/bryntum/support/issues/833). So for example we set for a resource `capacity` of `2` which make it collective and able to be allocated to 200%. This will allow to represent teams as resources and assign then but there won't be sub-resources so this use case won't work for everyone probably. Implementation is not that complex as well.

III. Having resources as a real tree brings up some new questions:

1. Assigning a group of resources should probably mean assigning of all the group members? Or we make such groups not assignable and use them for sake of visualization only (to group resources and roll up allocation in the histogram). 
2. If we assign groups then in real life membership in a group could change in time (a worker can be moved to another department for example). Which then makes the structure more complex (instead of one `resource` entity there should be 3 ones `resource <- membership -> group`)

**Comment by sdehandt** [comment]
Approach 1 would indeed already be a big help.
For approach 2, the capacity should also be calendar based. For example, it could be 2 but one the days that one team member has holidays, it should be set to 1 (manually by the user)

**Comment by arcady-zherdev** [comment]
> For approach 2, the capacity should also be calendar based. For example, it could be 2 but one the days that one team member has holidays, it should be set to 1 (manually by the user)

Yes, duly noted! And that makes option `#2` trickier. Need to think of it a bit more.

So far option `#1` looks the most clear on how it should work and can be done. So we'll start working on it I suppose.

**Comment by arcady-zherdev** [comment]
pushed initial effort to 2608-grouping-support branch

**Comment by sdehandt** [comment]
Any chance I can test a beta version of this?
